logs_sdk/lib/opentelemetry/sdk/logs/logger_provider.rb

```diff
@@ -34,14 +39,21 @@ def initialize(resource: OpenTelemetry::SDK::Resources::Resource.create)
         #
         # @return [OpenTelemetry::SDK::Logs::Logger]
         def logger(name:, version: nil)
-          version ||= ''
+          if @stopped
```

Comment:
Is this case necessary? I ask because it differs from what we're doing in the `TracerProvider`.

Reply:
It's been a minute since I wrote this, but I believe I did it to adhere to this part of the shutdown protocol: 

> `Shutdown` MUST be called only once for each `LoggerProvider` instance. After
the call to `Shutdown`, subsequent attempts to get a `Logger` are not allowed.
SDKs SHOULD return a valid no-op `Logger` for these calls, if possible. 

[(source)](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md?plain=1#L131-L133)

There's similar verbiage for [the Trace SDK](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md?plain=1#L132-L134). 

I'm not sure if we adhere to this part of the Trace spec elsewhere. Maybe this part does the job well enough? But I'm not sure if that would sufficiently cover an attempt to get a tracer after shutdown. 

https://github.com/open-telemetry/opentelemetry-ruby/blob/a1f1629abb7c87ae316cf0f4541111e48e433d67/sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb#L72-L75

What do you think?

Reply:
I think the other piece of the puzzle (for traces) is here: https://github.com/open-telemetry/opentelemetry-ruby/blob/a1f1629abb7c87ae316cf0f4541111e48e433d67/sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb#L146-L167

My interpretation of the spec is that after calling shutdown on a tracer provider all tracers issued by it, past or future, must be non-operational. That is, they will create no-op spans from there on out. They do not necessarily need to be an instance of a no-op tracer, if that makes sense.

I think we need to ensure that all loggers return early from `on_emit` after `shutdown` is called, then it doesn't really matter what kind of logger we return from the logger provider. For consistency, we can follow what we're doing for tracing. I'm open to other considerations though.

Reply:
I hadn't thought about it that way! Thanks for breaking that down. I've updated the code to use this approach. See: 06e4b05744729af78047a22dcbd1fa18b26a064d